softioc/builder.py

```diff
@@ -18,45 +18,52 @@
 # The SCAN field for all input records defaults to I/O Intr.
 
 
-def _in_record(record, name, **fields):
-    '''For input records we provide some automatic extra features: scanning,
-    initialisation as appropriate, and blocking puts from outside the IOC.'''
-
+def _set_in_defaults(fields):
     fields.setdefault('SCAN', 'I/O Intr')
-    if 'initial_value' in fields:
-        fields.setdefault('PINI', 'YES')
+    fields.setdefault('PINI', 'YES')
     fields.setdefault('DISP', 1)
-    return getattr(PythonDevice, record)(name, **fields)
+
+def _set_out_defaults(fields):
```

Comment:
What is the reason this isn't applied to `WaveformOut` and `longStringOut` records?

Reply:
The `waveform` record doesn't have an `OMSL` field, and as it happens that's the only field that `_set_out_defaults()` sets.

It's an anomaly, you'd expect an out `waveform` to be treated like any other Out record ... but in truth, it's _not_ an Out record at all, it's a `waveform` record (which can work in either direction) with special treatment.